src/wallet/mod.rs

```diff
@@ -299,7 +299,7 @@ where
             output: vec![],
         };
 
-        let fee_rate = builder.fee_rate.unwrap_or_default();
+        let fee_rate = get_fee_rate(&builder.fee_policy);
```

Comment:
I think the `fee_rate` should be set to `0.0` when there's an absolute fee (the default is `1.0`), otherwise in some edge cases the coin selection could fail because it would be considering the extra cost for each input added while in fact we don't care about it since we override the `fee_amount` later.

Also, the `fee_amount` should be overridden before the coin selection is done, so that the coin selection can properly take into account that amount of fees.